### PR TITLE
Add configuration support

### PR DESCRIPTION
**Pull request by ndench**
@phstc I've had a go at implementing configuration, however I've run into some problems I'm not sure how to solve. Hoping you can help me out :crossed_fingers: 

First up, I noticed that we can just read configuration directly from `Context` without the need for `probot-config`. [See the docs here](https://probot.github.io/api/latest/classes/context.html#config). So I tried that fist up because if we can do it without an extra dependency then that would be nice, [see my first commit](https://github.com/phstc/probot-merge-when-green/commit/98b8708f02fa0c9da2ba9aa5ecc03568d21ab50a). However I get some errors when running the tests:

```
    TypeError: context.config is not a function

      11 | 
      12 | export default async function getConfiguration (context: Context): Promise<IConfiguration> {
    > 13 |   return await context.config(CONFIGURATION_FILE, defaultConfig)
         |                        ^
      14 | }
      15 | 

      at Object.<anonymous> (src/configuration.ts:13:24)
      at step (src/configuration.ts:32:23)
      at Object.next (src/configuration.ts:13:53)
      at src/configuration.ts:7:71
      at Object.<anonymous>.__awaiter (src/configuration.ts:3:12)
      at Object.getConfiguration [as default] (src/configuration.ts:43:12)
      at Object.<anonymous> (src/mergeIfGreen.ts:19:40)
      at step (src/mergeIfGreen.ts:32:23)
      at Object.next (src/mergeIfGreen.ts:13:53)
      at fulfilled (src/mergeIfGreen.ts:4:58)
```

We're definitely running a version of Probot where the `Context.config` function exists so I'm not sure why it complains.

Then I thought I'd try `probot-config` because it just gives users an extra option. [See my second commit](https://github.com/phstc/probot-merge-when-green/commit/d95c4fbba3853cbabac5eca5bb5e2621be721d6d). But unfortunately I run into another problem with the tests:

```
    TypeError: Cannot read property 'getContents' of undefined

       9 | 
      10 | export default async function getConfiguration (context: Context): Promise<IConfiguration> {
    > 11 |   return await getConfig(context, CONFIGURATION_FILE)
         |                ^
      12 | }
      13 | 

      at loadYaml (node_modules/probot-config/lib/index.js:48:49)
      at getConfig (node_modules/probot-config/lib/index.js:114:24)
      at Object.<anonymous> (src/configuration.ts:11:16)
      at step (src/configuration.ts:32:23)
      at Object.next (src/configuration.ts:13:53)
      at src/configuration.ts:7:71
      at Object.<anonymous>.__awaiter (src/configuration.ts:3:12)
      at Object.getConfiguration [as default] (src/configuration.ts:41:12)
      at Object.<anonymous> (src/mergeIfGreen.ts:19:40)
      at step (src/mergeIfGreen.ts:32:23)
      at Object.next (src/mergeIfGreen.ts:13:53)
      at fulfilled (src/mergeIfGreen.ts:4:58)
```

I have no idea where to even start with this, so I'm hoping you've had experience with probot-config before?

And finally, when running the linter, I always get the following errors:

```
  src/configuration.ts:1:10: 'Context' is defined but never used.
  src/configuration.ts:6:18: 'IConfiguration' is not defined.
  src/configuration.ts:7:3: 'supportedCi' is not defined.
  src/configuration.ts:11:10: Redundant use of `await` on a return value.
```

These don't seem valid to me, because they're all definitely used and defined. Additionally if "solve" these linting errors by deleting what it's complaining about I get other problem with things not being defined (which is fully expected).

I've never done any work with probot before, and am fairly new to TypeScript too. So would appreciate some help :sweat_smile: 

Closes: #29

### CODE REVIEW

File: src/configuration.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  supportedCi: string[];
+}
+
+const defaultConfig: IConfiguration = {
+  supportedCi: ['circleci', 'travis-ci']
+}
+
+export default async function getConfiguration (context: Context): Promise<IConfiguration> {
+  return await context.config(CONFIGURATION_FILE, defaultConfig)
+}
```
Comment (phstc) [inline src/configuration.ts]: I loved this implementation without adding the extra dependency.

For making `yarn test` happy 👇 

```typescript
import { Context } from 'probot' // eslint-disable-line no-unused-vars
import { CONFIGURATION_FILE } from './constants'

const defaultConfig = {
  requiredChecks: ['circleci', 'travis-ci']
}

export default async function getConfiguration (context: Context): Promise<any> {
  return context.config(CONFIGURATION_FILE, defaultConfig)
}
```

(I changed from `supportedCi` to `requiredChecks`, WDYT?)

File: .github/merge-when-green.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+supportedCi:
+  - circleci
+  - travis-ci
+
+
```
Comment (phstc) [inline .github/merge-when-green.yml]: Sorry for taking long to get back to you.

BTW is this `.github/merge-when-green.yml` file needed if we stick with the first the commit implementation?

Would you mind to also update the README with some instructions for choosing "supportedChecks"?

Comment (ndench) [reply .github/merge-when-green.yml]: No problem @phstc, thanks for the feedback. No, this file is only needed if we have `probot-config`. If we do the first implementation then we have the `defaultConfiguration` object instead.

Hopefully I can finish this off over the weekend :+1: (including README update :wink:).

**Comment by ndench** [comment]
Thanks for the help @phstc, the tests now pass!

But I'm still getting a few linting errors, which aren't fixed by `yarn standard --fix`:

```
standard: Use JavaScript Standard Style (https://standardjs.com)
  src/configuration.ts:4:18: 'IConfiguration' is not defined.
  src/configuration.ts:5:3: 'requiredCi' is not defined.
  src/configuration.ts:15:69: 'IConfiguration' is not defined.
```

According to [this github comment](https://github.com/typescript-eslint/typescript-eslint/issues/131#issuecomment-483326781) we should not be using this rule on Typescript code because the compiler will pick these errors up anyway.

However, because we're using StandardJS we're [unable to ignore a rule](https://standardjs.com/index.html#i-disagree-with-rule-x-can-you-change-it).

To make the linter pass, we can remove all the type safety, like I did in [this reverted commit](https://github.com/phstc/probot-merge-when-green/pull/30/commits/157e2e5bf6b92cd52a7ba22b5d65524ec6d8a0d1). And it passes CI and is probably *fine*, I think type safety is good so I would rather have the interface.

Do you know how I can get this to pass? Or should we just remove the type safety?

**Comment by phstc** [comment]
> Do you know how I can get this to pass? Or should we just remove the type safety?

@ndench Thanks for looking into this. I would implement like this example 👇 the tests will pass.

https://github.com/phstc/probot-merge-when-green/pull/30/files#r305173117

**Comment by ndench** [comment]
@phstc ah right, sorry that's my bad. I didn't realise you'd already given me the answer! :joy: 
I was looking at other things you pointed out in that example.

It's all passing now, thanks for the help :+1:

**Comment by phstc** [comment]
Hey @ndench 

Great job! It worked like a charm. Thank you 🎉

**Pull request merged**